test/gtest/uct/test_ud_pending.cc

```diff
@@ -40,9 +40,12 @@ class test_ud_pending : public ud_base_test {
         }
     }
 
-    void check_pending_reqs(void) 
+    void check_pending_reqs(bool wait)
     {
-        /* all work should be complete */
+        /* wait for all work to be complete */
+        while (wait && (req_count < N)) {
```

Comment:
Maybe it is worth to introduce a timeout, like it is done in other tests to avoid hangs if something is not ok with pendings?